src/amazon-keyspaces-mcp-server/awslabs/amazon_keyspaces_mcp_server/server.py

```diff
@@ -0,0 +1,489 @@
+
+
+class DescribeKeyspaceInput(BaseModel):
+    """Input model for retrieving metadata for a specified keyspace."""
+
+    args: Dict[str, Any] = Field(
+        default_factory=dict, description='Arguments for the describe_keyspace operation'
+    )
+    keyspace: str = Field(..., description='The keyspace to retrieve metadata for.')
+
+
+class DescribeTableInput(BaseModel):
+    """Input model for retrieving metadata for a specified table."""
+
+    args: Dict[str, Any] = Field(
+        default_factory=dict, description='Arguments for the describe_table operation'
+    )
+    keyspace: str = Field(..., description='The keyspace containing the table')
+    table: str = Field(..., description='The name of the table to describe')
+
+
+class ExecuteQueryInput(BaseModel):
+    """Input model for executing a CQL query and returning the results."""
+
+    args: Dict[str, Any] = Field(
+        default_factory=dict, description='Arguments for the execute_query operation'
+    )
+    keyspace: str = Field(..., description='The keyspace to execute the query against')
+    query: str = Field(..., description='The CQL SELECT query to execute')
+
+
+class AnalyzeQueryPerformanceInput(BaseModel):
+    """Input model for heuristically analyzing the performance of a given CQL query."""
+
+    args: Dict[str, Any] = Field(
+        default_factory=dict, description='Arguments for the analyze_query_performance operation'
+    )
+    keyspace: str = Field(..., description='The keyspace to analyze the query against')
+    query: str = Field(..., description='The CQL query to analyze for performance')
+
+
+# Remove all default handlers then add our own
+logger.remove()
+logger.add(sys.stderr, level='INFO')
+
+mcp = FastMCP(name=SERVER_NAME, version=SERVER_VERSION)
+
+# Global handle to hold the proxy to the specific db client
+_proxy = None
+
+
+def get_proxy():
+    """Returns a singleton instance of the main Keyspaces MCP server implementation.
+
+    The singleton is initialized lazily.
+    """
+    global _proxy
+    if _proxy is None:
+        # Load configuration
+        app_config = AppConfig.from_env()
+
+        # Initialize client
+        cassandra_client = UnifiedCassandraClient(app_config.database_config)
+
+        # Initialize services
+        data_service = DataService(cassandra_client)
+        schema_service = SchemaService(cassandra_client)
+        query_analysis_service = QueryAnalysisService(cassandra_client, schema_service)
+
+        _proxy = KeyspacesMcpStdioServer(data_service, query_analysis_service, schema_service)
+
+    return _proxy
+
+
+@mcp.tool(
+    name='listKeyspaces',
+    description='Lists all keyspaces in the Cassandra/Keyspaces database - args: none',
+)
+def list_keyspaces(args: Optional[Dict[str, Any]] = None, ctx: Optional[Context] = None) -> str:
+    """Lists all keyspaces in the Cassandra/Keyspaces database."""
+    ListKeyspacesInput(**(args or {}))
+    return get_proxy().handle_list_keyspaces(args or {}, ctx)
```

Comment:
Can we add the Pydantic arg descriptions in the tool def (so the args will show up w descriptions on the tool definition) instead of separating it out to a separate class? 
```suggestion
@mcp.tool(
    name='listKeyspaces',
    description='Lists all keyspaces in the Cassandra/Keyspaces database - args: none',
)
def list_keyspaces(args: Dict[str, Any] = Field(
        default_factory=dict, description='Arguments for the list_keyspaces operation'),
    ctx: Optional[Context] = None) -> str:
    """Lists all keyspaces in the Cassandra/Keyspaces database."""
    return get_proxy().handle_list_keyspaces(args or {}, ctx)
```





    args: Dict[str, Any] = Field(
        default_factory=dict, description='Arguments for the list_tables operation'
    )
    keyspace: str = Field(..., description='The keyspace to list tables from')